lib/puppet/pops/binder/lookup.rb

```diff
@@ -1,196 +1,90 @@
 # This class is the backing implementation of the Puppet function 'lookup'.
-# See puppet/parser/functions/lookup.rb for documentation.
+# See puppet/functions/lookup.rb for documentation.
 #
 class Puppet::Pops::Binder::Lookup
+  # Performs a lookup in the configured scopes and optionally merges the default.
+  #
+  # This is a backing function and all parameters are assumed to have been type checked.
+  # See puppet/functions/lookup.rb for full documentation and all parameter combinations.
+  #
+  # @param scope [Puppet::Parser::Scope] The scope to use for the lookup
+  # @param name [String|Array<String>] The name or names to lookup
+  # @param type [String|Puppet::Pops::Types::PAnyType] The expected type of the found value. Can be nil
+  # @param default_value [Object] The value to use as default when no value is found
+  # @param accept_undef [Boolean] true if it's accepable to return nil when no value is found. Can be nil
+  # @param override [Hash<String,Object>] A map to use as override. Values found here are returned immediately. Can be empty.
```

Comment:
maybe just add (no merge) in this short note, or (see description) - hard to read when I was assumed to know what "immediate return" implied.